### PR TITLE
CUMULUS-3718: Update Reconciliation Report List endpoints to query postgres

### PR DESCRIPTION
**Pull request by charleshuang80**
**Summary:** Summary of changes

Addresses [CUMULUS-3718: Update Reconciliation Report List endpoints to query postgres](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3718)

## Changes

* add db/search/ReconciliationReportSearch
* make api list endpoint for recon reports work using the added search
* fix/clean test file for recon reports endpoints - list test works, remove ES references
* add recon reports to stats endpoint
* add tests for recon reports stats endpoint
* add test for recon reports aggregate in test-StatsSearch

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/db/tests/search/test-ReconciliationReportSearch.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    infix: 'conReport-2',
+  }
+  let dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  let response = await dbSearch.query(knex);
+  t.is(response.meta.count, 11);
+  t.is(response.results?.length, 11);
+  
+  // queryStringParameters = {
+  //   limit: 50,
+  //   infix: 'ending', // ending, status
```
Comment (charleshuang80) [inline packages/db/tests/search/test-ReconciliationReportSearch.js:139]: In [the api docs](https://nasa.github.io/cumulus-api/unreleased/#cumulus-api), it says for infix `includes search of the granuleId, status, pdrName, name, error, id and msg fields`. But this search I tried for infix in `status` didn't work. When I looked at what I copied from other search classes for `buildInfixPrefixQuery`, it only looks at "query.whereLike(`${this.tableName}.name`, `%${infix}%`)", so do I need to adjust the class, or is it only really supposed to look at `name`?

File: packages/api/tests/endpoints/test-stats.js [hunk 2]
```diff
@@ -100,15 +104,18 @@ test.before(async (t) => {
     }))
   ));
 
-  await t.context.collectionPgModel.insert(
-    t.context.knex,
-    collections
-  );
+  range(24).map((num) => (
```
Comment (etcart) [inline packages/api/tests/endpoints/test-stats.js:107]: nit: I think this should be foreach since we're not returning anything from it

File: packages/db/tests/search/test-ReconciliationReportSearch.js (hunk 1 above)
Comment (etcart) [reply packages/db/tests/search/test-ReconciliationReportSearch.js:139]: as I understood this, infix query is for parsing something specific to the table in question (in this case name), but other, more generally present fields like granuleId are handled elsewhere in the buildSearch function (line 116 of BaseSearch.ts) by this.buildTermQuery mainly.  infix is just the one we need to tailor to this table

File: packages/api/tests/endpoints/test-stats.js (hunk 2 above)
Comment (charleshuang80) [reply packages/api/tests/endpoints/test-stats.js:107]: hmm... I did it this way because so many of the other tests are doing it the same way

File: packages/db/tests/search/test-ReconciliationReportSearch.js (hunk 1 above)
Comment (charleshuang80) [reply packages/db/tests/search/test-ReconciliationReportSearch.js:139]: so it's fine if we are supposed to customize/specify the field that the infix works for per table. But that isn't what our docs suggest, the docs seem to suggest you can do an includes search for any of the listed fields.
As an aside, one other issue I have with the docs for infix (maybe prefix also) and many of the other searches we are testing is that we don't have examples showing that some of those searches are possible.

Comment (etcart) [reply packages/db/tests/search/test-ReconciliationReportSearch.js:139]: hm, I guess *is* that a behavior of the elasticsearch Search functionality we're imitating? in which case the docs were always wrong and should probably be fixed. or is that a functionality we've been failing to replicate so far in these moves over to PG

Comment (charleshuang80) [reply packages/db/tests/search/test-ReconciliationReportSearch.js:139]: I'm not sure, but it seems to me either way our docs are a bit off

Comment (jennyhliu) [reply packages/db/tests/search/test-ReconciliationReportSearch.js:139]: With ElasticSearch, we can easily match multiple fields. If we support the same infix/prefix search in postgres, the query can be complicated and performance is an concern, so we only support match one column per table.  If it becomes necessarily to match multiple columns, we can update in the future.

File: packages/api/tests/endpoints/test-stats.js (hunk 2 above)
Comment (charleshuang80) [reply packages/api/tests/endpoints/test-stats.js:107]: I think if I make this change, then to be consistent I need to make a similar change to the other data types here (collections, granules) and in most of these other test files (endpoint tests, search tests). Is that what you think I should do?

Comment (jennyhliu) [reply packages/api/tests/endpoints/test-stats.js:107]: You can just update the files you added/updated.  The original implementation is not wrong, just not as simple.

Comment (charleshuang80) [reply packages/api/tests/endpoints/test-stats.js:107]: I can do that, but I feel like we should have some way of noting that if anyone makes changes to the other files, they should make a similar change...

**Review by jennyhliu — APPROVED** [review]
Good work! Thank you

File: audit-ci.json [hunk 3]
```diff
@@ -3,12 +3,6 @@
   "pass-enoaudit": true,
   "retry-count": 20,
   "allowlist": [
-    {
```
Comment (Jkovarik) [inline audit-ci.json:6]: Why was this removed?

Comment (charleshuang80) [reply audit-ci.json:6]: It was expired, so didn't seem like it was needed anymore.
